AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -457,15 +457,27 @@ - (void)_lazilyLoadImageIfNecessary
           } else {
             // First try to load the path directly, for efficiency assuming a developer who
             // doesn't want caching is trying to be as minimal as possible.
-            self.image = [UIImage imageWithContentsOfFile:_URL.path];
-            if (!self.image) {
+            NSData *data = [NSData dataWithContentsOfURL:_URL];
+            if (data == nil) {
               // If we couldn't find it, execute an -imageNamed:-like search so we can find resources even if the
               // extension is not provided in the path.  This allows the same path to work regardless of shouldCacheImage.
               NSString *filename = [[NSBundle mainBundle] pathForResource:_URL.path.lastPathComponent ofType:nil];
-              if (filename) {
-                self.image = [UIImage imageWithContentsOfFile:filename];
+              if (filename != nil) {
+                data = [NSData dataWithContentsOfFile:filename];
               }
             }
+
+            // If the file may be an animated gif and then created an animated image.
+            id<ASAnimatedImageProtocol> animatedImage = nil;
+            if (_downloaderImplementsAnimatedImage && data != nil && [_URL.pathExtension isEqualToString:@"gif"]) {
+              animatedImage = [_downloader animatedImageWithData:data];
+            }
+
+            if (animatedImage != nil) {
+              self.animatedImage = animatedImage;
+            } else {
+              self.image = [UIImage imageWithData:data];
```

Comment:
This may not have any caching behavior, but UIImage does funny things.  I'm wondering if this does any form of caching that ...contentsoffile does not do.  @garrettmoon how about that internal UIImage decoded image cache that we had to disable with CG options?  Is that important for animated images?  I think even if it only caches the last frame drawn, it's relevant, because decoded image memory is big.